modules/agent/pkg/register/register.go

```diff
@@ -187,13 +186,8 @@ func createClusterSecret(ctx context.Context, clusterID string, k8s corecontroll
 		newToken := newSecret.Data[Token]
 		clusterNamespace := newSecret.Data[ClusterNamespace]
 		clusterName := newSecret.Data[ClusterName]
-		systemNamespace := string(newSecret.Data[SystemNamespace])
 		deploymentNamespace := newSecret.Data[DeploymentNamespace]
 
-		if !cfg.IgnoreAgentNamespaceCheck && systemNamespace != secret.Namespace {
-			return nil, fmt.Errorf("fleet-agent must be installed in the namespace %s", systemNamespace)
-		}
```

Comment:
Never checking the namespace seems fine to me, though I wonder what was the motivation for its original addition.

Reply:
It was that 90% of the plumbing was there to support this scenario of not having the namespaces matching, but it wasn't complete, so we ensured the namespaces matched.  Now in theory it should work and we should be implicitly testing it in rancher.

Reply:
That makes sense to me. Thanks.